### ISSUE TITLE
ble_monitor receives value only once after submit and throws exception

### ISSUE THREAD

**Issue by archiespb** [custom-components/ble_monitor#683]
I have clean installation (after many tries) of Home Assistant in VM (TrueNAS Scale is host OS) with HACS and Passive BLE Monitor.
System has two bluetooth devices - Intel (internal) and dongle.
In both cases (regardless of selected bluetooth interface) BLE monitor receives data from ble devices once, throws  one of two errors and retrieves no more data until system reboot or Submit button pressing.

Errors are:
```
2022-02-01 07:13:16 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/config/custom_components/ble_monitor/sensor.py", line 234, in async_run
entity = sensors[device_sensors.index(measurement)]
IndexError: list index out of range
```
or
```
2022-02-01 19:46:57 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/config/custom_components/ble_monitor/sensor.py", line 235, in async_run
entity.collect(data, period_cnt, batt_attr)
File "/config/custom_components/ble_monitor/sensor.py", line 516, in collect
self._measurements.append(data[self.entity_description.key])
KeyError: 'voltage'
```

BLE devices are CGG1, two different versions from ClearGrass and Qingping, one of them - with ATC firmware (unfortunately I can't recall which of them). 
There are same errors on raspberry pi model 3 b+ with internal bluetooth and external dongle.
I tried many times to install, remove, reinstall ble_monitor, HACS and Home Assistant at all and every time there is an error described.

**Comment by Ernst79** [comment]
On the ATC one, which advertisement formats did you set it to send?

**Comment by Ernst79** [comment]
Yes, probably because it restores the wrong list with supported sensors upon restart. Could you delete the sensor from the BLE monitor options menu (select MAC and delete it)? It will come back, hopefully with the correct list

**Comment by archiespb** [comment]
Sadly (((
```
Logger: homeassistant
Source: custom_components/ble_monitor/sensor.py:516
Integration: Passive BLE monitor (documentation, issues)
First occurred: 9:45:10 PM (1 occurrences)
Last logged: 9:45:10 PM

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/ble_monitor/sensor.py", line 235, in async_run
    entity.collect(data, period_cnt, batt_attr)
  File "/config/custom_components/ble_monitor/sensor.py", line 516, in collect
    self._measurements.append(data[self.entity_description.key])
KeyError: 'voltage'
```

**Comment by archiespb** [comment]
I set advertisement type to "Mi" and error repeats.
Now I set advertisement type to "ATC1441" and there are no error about 5 minuutes. BLE monitor appropriately shows temperature and humidity changes.

**Comment by Ernst79** [comment]
But do you have an voltage sensor with "unknown" as state?

**Comment by archiespb** [comment]
Its name is ATC and it has a working "voltage" sensor. But I didn't delete "voltage" from line 762 in const.py.

![image](https://user-images.githubusercontent.com/43927989/152034751-ea20fc7e-a293-4f2f-b217-05ef27727dc3.png)
![image](https://user-images.githubusercontent.com/43927989/152034874-4ea8c0cb-36a4-4c8d-83f5-403167be84ec.png)

**Comment by Ernst79** [comment]
Ah, of course. ATC (custom) is correct. 

But if you now set it back to "All", the error return?

**Comment by archiespb** [comment]
Yes, while advertisement type set to All error returns.

**Comment by Ernst79** [comment]
could you try 7.4.1-beta, just to be sure.

**Comment by archiespb** [comment]
Same error.

```
Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/ble_monitor/sensor.py", line 235, in async_run
    entity.collect(data, period_cnt, batt_attr)
  File "/config/custom_components/ble_monitor/sensor.py", line 516, in collect
    self._measurements.append(data[self.entity_description.key])
KeyError: 'voltage'
```

**Comment by Ernst79** [comment]
Could you enable debug logging for ble_monitor (at debug level) and post the lines in the log right before this error occurs. There should be a line with the advertisement it has received, probably without voltage.

**Comment by archiespb** [comment]
Sure.

```
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.device_tracker] Starting device tracker entry startup
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.device_tracker] BLE device tracker updater initialization
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.device_tracker] BLE device tracker updater initialized
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.device_tracker] Device Tracker entry setup finished
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.device_tracker] Device tracker updater loop started!
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Data measuring sensor received: {'temperature': 25.38, 'humidity': 36.82, 'voltage': 3.015, 'battery': 90, 'switch': 0, 'opening': 1, 'status': 'opened', 'data': True, 'rssi': -79, 'mac': '5A582D34125E', 'type': 'ATC', 'packet': 30, 'firmware': 'ATC (Custom)'}
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble rssi 087CBE84211B
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble temperature 087CBE84211B
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble temperature 582D3410EDE5
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble humidity 582D3410EDE5
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble battery 582D3410EDE5
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble voltage 582D3410EDE5
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble rssi 582D3410EDE5
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble temperature 5A582D34125E
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble humidity 5A582D34125E
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble battery 5A582D34125E
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble voltage 5A582D34125E
2022-02-03 19:06:53 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble rssi 5A582D34125E
2022-02-03 19:06:57 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Data measuring sensor received: {'rssi': -78, 'mac': '5A582D34125E', 'type': 'CGG1-ENCRYPTED', 'packet': 31, 'firmware': 'Xiaomi (MiBeacon V3)', 'data': True, 'temperature': 25.4, 'humidity': 34.0}
2022-02-03 19:06:57 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/ble_monitor/sensor.py", line 235, in async_run
    entity.collect(data, period_cnt, batt_attr)
  File "/config/custom_components/ble_monitor/sensor.py", line 516, in collect
    self._measurements.append(data[self.entity_description.key])
KeyError: 'voltage'
```

**Comment by Ernst79** [comment]
Thanks, found it. It was CGG1-ENCRYPTED, not CGG1. Fix will be in the next release tonight.

**Comment by Ernst79** [comment]
Fixed in 7.5.1
